### PR TITLE
[single] Implementation of flexible input dimension

### PR DESCRIPTION
**Pull request by kparichay**
Added implementation of flexible input dimension at the API level
Handling flexible input dimension at the tensor filter extension
scope to be done.

Check #1651 for more information

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor pk.kapoor@samsung.com

### CODE REVIEW

File: api/capi/src/nnstreamer-capi-single-new.c [hunk 1]
```diff
@@ -732,6 +734,27 @@ ml_single_set_timeout (ml_single_h single, unsigned int timeout)
...
+
+  if (!single)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  ML_SINGLE_GET_VALID_HANDLE_LOCKED (single_h, single, 0);
+
+  switch (single_h->nnfw) {
+    case ML_NNFW_TYPE_TENSORFLOW_LITE:
+      klass = g_type_class_peek (G_TYPE_TENSOR_FILTER_SINGLE);
+      if (klass->set_input_info (single_h->filter, info) == FALSE)
```
Comment (jaeyun-jung) [inline api/capi/src/nnstreamer-capi-single-new.c]: TODO fix possible svace issue later
if (klass == NULL || klass->set_input_info (single_h->filter, info) == FALSE)

File: api/capi/src/nnstreamer-capi-single-new.c [hunk 2]
```diff
@@ -450,6 +451,7 @@ ml_single_open (ml_single_h * single, const char *model,
    * 3. Construct a pipeline
    * Set the pipeline desc with nnfw.
    */
+  single_h->nnfw = nnfw;
```
Comment (dongju-chae) [inline api/capi/src/nnstreamer-capi-single-new.c]: Is there any possibility to overwrite `single_h->nnfw` when this open API is called twice?
Then, it's better to check this at the beginning.

Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: With every open to `ml_single_open`, a new `single_h` object is made. So, if this open API is called twice, two `single_h` objects will be allocated and returned to the user.
So, I think overwrite should not happen.

Comment (dongju-chae) [reply api/capi/src/nnstreamer-capi-single-new.c]: Oh, okay

File: api/capi/src/nnstreamer-capi-single-new.c (hunk 1 above)
Comment (kparichay) [reply api/capi/src/nnstreamer-capi-single-new.c]: @jaeyun-jung https://github.com/nnsuite/nnstreamer/pull/1820

**Pull request merged**